### DISCUSSION TITLE
Is there any way to use Extension or State in non-handler functions without passing to them?

### DISCUSSION THREAD

**Discussion by hlf20010508** [tokio-rs/axum#2764] [Q&A] [answered]
### Summary

Is there a function like `Extension::get::<Arc<MyData>>()` that I can use out side of handlers?
I seperated many functions from handlers. I don't want to pass those extensions and state one by one.

### axum version

0.7.5

**Comment by mladedav** [comment, marked as answer]
Not really since the function would have to return different things for different requests and in the case of state different things for different handlers.

So you need a raw `Request` (or rather its extensions) and you could pass around that if you're using a lot of them.
